### PR TITLE
LE Audio: ASCS release state bug fix

### LINKED ISSUE #44621: ASCS: Sink ASE stuck in Releasing state
**Describe the bug**
The Sink ASE stuck in Releasing state. No transition to either Idle or Codec Configured state is made.
Note that CIS has been disconnected already by remote.

**To Reproduce**
* Let remote client to Configure and Start Stream
* Disable ASE as remote client
* Observe ASE state transition to QoS configured state
* Release ASE as remote client
* Observe ASE state transition to Relasing state and no further transition to Idle/Codec Configured state

**Expected behavior**
As per spec:
"If the server accepts the requested Release operation parameter values for an ASE, or the server
autonomously initiates the Release operation for an ASE, the server shall transition the ASE to the
Releasing state and write a value of 0x06 (Releasing) to the ASE_State field."

**Impact**
Remote client cannot use the ASE as only Server can transit ASE from Releasing state.
image
image

**Logs and console output**
```
> ACL Data RX: Handle 64 flags 0x02 dlen 10                                                                                                                                    #294 57.307928
      ATT: Write Command (0x52) len 5
        Handle: 0x001c
          Data: 050101
        ASCS: Audio Stream Endpoint Control Point (Handle: 0x001c)
        Disable
        Number of ASEs: 1
        ASE ID: 1
< HCI Command: Host Number of Completed Packets (0x03|0x0035) plen 5                                                                                                           #295 57.312788
        Num handles: 1
        Handle: 64
        Count: 1
< ACL Data TX: Handle 64 flags 0x00 dlen 12                                                                                                                                    #296 57.313131
      ATT: Handle Value Notification (0x1b) len 7
        Handle: 0x001c
          Data: 0501010000
        ASCS: Audio Stream Endpoint Control Point (Handle: 0x001c)
        Operation response: Disable
        Number of ASEs: 1
         ASE ID: 1
          Response code: Success
< ACL Data TX: Handle 64 flags 0x00 dlen 24                                                                                                                                    #297 57.314297
      ATT: Handle Value Notification (0x1b) len 19
        Handle: 0x0016
          Data: 0102020310270000023c00020a00204e00
        ASCS: Audio Stream Sink Endpoint (Handle: 0x0016)
        ASE ID: 1
        QoS configured
          CIG ID: 2
          CIS ID: 3
          SDU interval: 10000
          Framing: Unframed
          LE PHY (0x02)
            LE 2M PHY
          Maximum SDU size: 60
          Retransmission number: 2
          Transport latency: 10 ms
          Presentation delay: 20000
> ISO Data RX: Handle 102 plen 72                                                                                                                                              #298 57.323934
        Packet boundary flag: 0x02
        Timestamp flag: 1
        ISO data length: 68
          Timestamp: 272125358
          Packet seq. nb.: 105
          SDU length: 60
          Packet status flags: 0x00
(...)
> ISO Data RX: Handle 102 plen 12                                                                                                                                              #313 57.451979
        Packet boundary flag: 0x02
        Timestamp flag: 1
        ISO data length: 8
          Timestamp: 272095359
          Packet seq. nb.: 119
          SDU length: 0
          Packet status flags: 0x02
> HCI Event: Disconnect Complete (0x05) plen 4                                                                                                                                 #314 57.451985
        Status: Success (0x00)
        Handle: 102
        Reason: Remote User Terminated Connection (0x13)
< ACL Data TX: Handle 64 flags 0x00 dlen 24                                                                                                                                    #315 57.457182
      ATT: Handle Value Notification (0x1b) len 19
        Handle: 0x0016
          Data: 0102020310270000023c00020a00204e00
        ASCS: Audio Stream Sink Endpoint (Handle: 0x0016)
        ASE ID: 1
        QoS configured
          CIG ID: 2
          CIS ID: 3
          SDU interval: 10000
          Framing: Unframed
          LE PHY (0x02)
            LE 2M PHY
          Maximum SDU size: 60
          Retransmission number: 2
          Transport latency: 10 ms
          Presentation delay: 20000
> HCI Event: Number of Completed Packets (0x13) plen 5                                                                                                                         #316 57.579974
        Num handles: 1
        Handle: 64
        Count: 1
> ACL Data RX: Handle 64 flags 0x02 dlen 10                                                                                                                                    #317 69.931722
      ATT: Write Command (0x52) len 5
        Handle: 0x001c
          Data: 080101
        ASCS: Audio Stream Endpoint Control Point (Handle: 0x001c)
        Release
        Number of ASEs: 1
        ASE ID: 1
< HCI Command: Host Number of Completed Packets (0x03|0x0035) plen 5                                                                                                           #318 69.936628
        Num handles: 1
        Handle: 64
        Count: 1
< ACL Data TX: Handle 64 flags 0x00 dlen 12                                                                                                                                    #319 69.936760
      ATT: Handle Value Notification (0x1b) len 7
        Handle: 0x001c
          Data: 0801010000
        ASCS: Audio Stream Endpoint Control Point (Handle: 0x001c)
        Operation response: Release
        Number of ASEs: 1
         ASE ID: 1
          Response code: Success
< ACL Data TX: Handle 64 flags 0x00 dlen 9                                                                                                                                     #320 69.938219
      ATT: Handle Value Notification (0x1b) len 4
        Handle: 0x0016
          Data: 0106
        ASCS: Audio Stream Sink Endpoint (Handle: 0x0016)
        ASE ID: 1
        Releasing
> HCI Event: Number of Completed Packets (0x13) plen 5                                                                                                                         #321 69.995729
        Num handles: 1
        Handle: 64
        Count: 2
> ACL Data RX: Handle 64 flags 0x02 dlen 7                                                                                                                                     #322 72.939619
      ATT: Read Request (0x0a) len 2
        Handle: 0x0016
        ASCS: Audio Stream Sink Endpoint Value (Handle: 0x0016)
< HCI Command: Host Number of Completed Packets (0x03|0x0035) plen 5                                                                                                           #323 72.942003
        Num handles: 1
        Handle: 64
        Count: 1
< ACL Data TX: Handle 64 flags 0x00 dlen 7                                                                                                                                     #324 72.942027
      ATT: Read Response (0x0b) len 2
        Value: 0106
        ASCS: Audio Stream Sink Endpoint (Handle: 0x0016)
        ASE ID: 1
        Releasing
> HCI Event: Number of Completed Packets (0x13) plen 5                                                                                                                         #325 73.051619
        Num handles: 1
        Handle: 64
        Count: 1
```
Console output and HCI traces attached
[ascs_releasing_state.log](https://github.com/zephyrproject-rtos/zephyr/files/8440735/ascs_releasing_state.log)
[ascs_releasing_state_hci.log](https://github.com/zephyrproject-rtos/zephyr/files/8440736/ascs_releasing_state_hci.log)

### PR DESCRIPTION
**Pull request by Thalley**
Fixes a bug for the ASCS release state, and updates the BSIM to also test the release procedure. 

Fixes https://github.com/zephyrproject-rtos/zephyr/issues/44621

### CODE REVIEW

File: subsys/bluetooth/audio/ascs.c [hunk 1]
```diff
@@ -133,13 +133,15 @@ void ascs_ep_set_state(struct bt_audio_ep *ep, uint8_t state)
 				ops->disabled(stream);
 			}
 			break;
+		case BT_AUDIO_EP_STATE_RELEASING:
```
Comment (asbjornsabo) [inline subsys/bluetooth/audio/ascs.c:136]: Indentation seems to be one or two TABs off, from line 134, - may be just Github, but please check.

**Review by asbjornsabo — APPROVED** [review]
LGTM - with the exception of a possible indentation issue.

File: subsys/bluetooth/audio/ascs.c (hunk 1 above)
Comment (Thalley) [reply subsys/bluetooth/audio/ascs.c:136]: Just Github :)

**Review by MariuszSkamra — APPROVED** [review]
LGTM, tested

**Pull request merged**